script-indent does not work properly with the ternary operator

<!--
    This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

    1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
    2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes

    Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:** 5.8.0
* **eslint-plugin-vue Version:** 5.0.0-beta3
* **Node Version:** 11

**Please show your full configuration:**

<!-- Paste your configuration below: -->
```js
module.exports = {
  extends: [
    'plugin:vue/recommended',
  ],

  parserOptions: {
    parser: 'babel-eslint',
  },

  overrides: [
    // Disable the default `indent` rule for `.vue` files to avoid conflicts
    // with `vue/script-indent`.
    { files: ['*.vue'], rules: { indent: 'off' } },
  ],
};
```

**What did you do? Please include the actual source code causing the issue.**

<!-- Paste the source code below: -->
```vue
<template>
  <img :src="imgUrl">
</template>

<script>
export default {
  data() {
    const hours = new Date().getHours();

    return {
      imgUrl: hours > 21 && hours < 6 ? 'night.png' :
        'day.png',
    };
  },
}
</script>
```

**What did you expect to happen?**

I expect the code to be successfully linted.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  40:1  error  Expected indentation of 6 spaces but found 8 spaces  vue/script-indent
```

**Additional notes**

I was able to reproduce the issue in the playground: [follow this link](https://mysticatea.github.io/vue-eslint-demo/#eJyNVjuTnDgQ/isqAu9u1ciuc3DBeG0nF1x+d9kmAhqQV0hYD2bw1vz3a0kgBMxWOZga9HWr1Y+vW3orKlVDcS6eLfSDYBa+vUhCnnnfkrPR1deXAj//0+KlQMHzp0wLl6bSfLD4DddBaUtqaJgTlrx5GzWz7PEpfhNSKWks6ZTThnwlEi7kLzTz+PSxBfu3Rx+fvnibXleDdVouOwmJHpzn3d/I5z/Ihw/z6pn8Sb6TB8nbzn4cZPtAzss2Qh5qNgXwFLHbF/9/w9XNBzO7X5wK7QSY4vxWBDe1q6zS1LgBdHH+fCoaXNVcQ2W5kgFBry1oGj0NiFS0YgZoDZVgmnlNkwSCGUPxx9tVWfUD00AltPQXaLXisj6oSqME5Et7R8UyacN2ntyMEquVQFdbuC5gDaVr2zm6sBZggY5sRTB4ynRrNkAMpIe+BL2VvMKUA4JXWN6QkQVF5tiJVh0mp/K5C7a2woH5rMoNmBbXXchwtZrRUmFmmPQn2a3EQM8XpGFC2E4r13YJcrLaWWyFKpnYgVxKdPZeVbkcmeB1zOyQUI1rh8r00nELZmBVSkHPr1DTACEdsM6Wlckc9gQ1U18qsSCq/IFxCZFUVGVZkmqITsEKoB+z9QUzIJpdQKigja+tZmvJbMcNLQGJDhnvUeAkNvW6wECxCzCIHvucCy5hlWlgWN1SZJBhDdCGSwxi2qHIe5bz1Eln0K5gJawBzyDycoUwJEASgqnYEI/CPKFT1HQ4gzpMawA1/HTYsnTiICKCOyk3ksUjY+3sNICK8Y0OPiH/NC8dMrebhg7k6uFGaqjS9ZwjL8Be7gGbb54RI+SSQUkvk6zH4yVdJqhnLJdt0uxsL7AnlAdpiT3yCtZzIuX4XSVf8YMlCOya2zehHKsp7Rb76ZSFnV6gzXxOkvwwV599s5bDoz3DxlwTg9ShG5cTUWj0S0BIlR9LIS27CHuHc6zqfKpwFAxKhBIcYgzZHLTC4/xQ2aYSWcLMhO2N6fYFcJ6wszLPQo3TsXEhmyP1U36kvMnl29m2gvN8WwPP5SHkvBFn3DeePwq0VjoXaDCgR3Ryf5LBaYnwyiJ/qW4UeA0UmgZpZ34j3GX04CREfmCH4LjzsyHsTcHkOxJd0bW7OLpYq8tWdLXoNqNLLXPh3NEpInNHuOFXbPlQGHrhtouFysVjYFZCQmsuudid4bMSO3FHmWVarJnm5iCcnzchtwdhMO3HCc3eEO9rHa1rbNntkyKXRnrmRUiSMMje8c3buseL1Ux8BO0ng0+OgN9v23B/ZNMtCUZaomlq7CRy0Lf0ForjeGGVVsruZNHSAcQpxV4PKN4hkLdyjh/AnE8LtqHUAt4x2OPz+aiYXRorVB1PHvQRw2vscgB9TyF4OxWxUP/wX3FnXP47+auwiI+NUxFueIypwOeHBihu/wOmvxW4)

Additionally, if I change the JS code from:

```js
imgUrl: hours > 21 && hours < 6 ? 'night.png' :
  'day.png',
```

to

```js
imgUrl: hours > 21 && hours < 6 ?
  'night.png' : 'day.png',
```

The the lint validation succeeds. [Here is a playground link for a fixed case](https://mysticatea.github.io/vue-eslint-demo/#eJyNVjuTnDgQ/isqAu9u1ciuc3DBeG0nF1x+d9kmAhqQV0hYD2bw1vz3a0kgBMxWOZga9HWr1Y+vW3orKlVDcS6eLfSDYBa+vUhCnnnfkrPR1deXAj//0+KlQMHzp0wLl6bSfLD4DddBaUtqaJgTlrx5GzWz7PEpfhNSKWks6ZTThnwlEi7kLzTz+PSxBfu3Rx+fvnibXleDdVouOwmJHpzn3d/I5z/Ihw/z6pn8Sb4vioQ8SN529uMg2wdyJg81m8L3KWrcvvj/G65uPpjZ/eJUaCfAFOe3IripXWWVpsYNoIvz51PR4KrmGirLlQwIem1B0+hpQKSiFTNAa6gE08xrmiQQzBiKP96uyqofmAYqoaW/QKsVl/VBVRolIF/aOyqWSRu28+RmlFitBLrawnUBayhd287RhbUAC3RkK4LBU6ZbswFiID30Jeit5BWmHBC8wvKGjCwoMsdOtOowOZXPXbC1FQ7MZ1VuwLS47kKGq9WMlgozw6Q/yW4lBnq+IA0TwnZaubZLkJPVzmIrVMnEDuRSorP3qsrlyASvY2aHhGpcO1Sml45bMAOrUgp6foWaBgjpgHW2rEzmsCeomfpSiQVR5Q+MS4ikoirLklRDdApWAP2YrS+YAdHsAkIFbXxtNVtLZjtuaAlIdMh4jwInsanXBQaKXYBB9NjnXHAJq0wDw+qWIoMMa4A2XGIQ0w5F3rOcp046g3YFK2ENeAaRlyuEIQGSEEzFhngU5gmdoqbDGdRhWgOo4afDlqUTBxER3Em5kSweGWtnpwFUjG908An5p3npkLndNHQgVw83UkOVrucceQH2cg/YfPOMGCGXDEp6mWQ9Hi/pMkE9Y7lsk2Zne4E9oTxIS+yRV7CeEynH7yr5ih8sQWDX3L4J5VhNabfYT6cs7PQCbeZzkuSHufrsm7UcHu0ZNuaaGKQO3biciEKjXwJCqvxYCmnZRdg7nGNV51OFo2BQIpTgEGPI5qAVHueHyjaVyBJmJmxvTLcvgPOEnZV5Fmqcjo0L2Rypn/Ij5U0u3862FZzn2xp4Lg8h5404477x/FGgtdK5QIMBPaKT+5MMTkuEVxb5S3WjwGug0DRIO/Mb4S6jBych8gM7BMednw1hbwom35Hoiq7dxdHFWl22oqtFtxldapkL545OEZk7wg2/YsuHwtALt10sVC4eA7MSElpzycXuDJ+V2Ik7yizTYs00Nwfh/LwJuT0Ig2k/Tmj2hnhf62hdY8tunxS5NNIzL0KShEH2jm/e1j1erGbiI2g/GXxyBPx+24b7I5tuSTDSEk1TYyeRg76lt1AcxwurtFJ2J4uWDiBOKfZ6QPEOgbyVc/wA5nxasA2lFvCOwR6fz0fF7NJYoep48qCPGF5jlwPoewrB26mIhfqH/4o74/LfyV+FRXxsnIpww2NMBT4/NEBx+x+XDhW4)

--- a/lib/utils/indent-common.js
+++ b/lib/utils/indent-common.js
@@ -984,12 +984,16 @@ module.exports.defineVisitor = function create (context, tokenStore, defaultOpti
     },
 
     ConditionalExpression (node) {
+      const prevToken = tokenStore.getTokenBefore(node)
       const firstToken = tokenStore.getFirstToken(node)
       const questionToken = tokenStore.getTokenAfter(node.test, isNotRightParen)
       const consequentToken = tokenStore.getTokenAfter(questionToken)
       const colonToken = tokenStore.getTokenAfter(node.consequent, isNotRightParen)
       const alternateToken = tokenStore.getTokenAfter(colonToken)
-      const isFlat = (node.test.loc.end.line === node.consequent.loc.start.line)
+      const isFlat =
+        prevToken &&
+        prevToken.loc.end.line !== node.loc.start.line &&
+        node.test.loc.end.line === node.consequent.loc.start.line
 
       if (isFlat) {
         setOffset([questionToken, consequentToken, colonToken, alternateToken], 0, firstToken)

--- a/tests/fixtures/script-indent/issue625.vue
+++ b/tests/fixtures/script-indent/issue625.vue
@@ -1,0 +1,12 @@
+<!--{}-->
+<script>
+export default {
+  data() {
+    const hours = new Date().getHours();
+    return {
+      imgUrl: hours > 21 && hours < 6 ? 'night.png' :
+        'day.png',
+    };
+  },
+}
+</script>

